app/controllers/workflow_executions/submissions_controller.rb

```diff
@@ -3,16 +3,23 @@
 module WorkflowExecutions
   # Workflow submission controller
   class SubmissionsController < ApplicationController
+    include Metadata
     respond_to :turbo_stream
     before_action :workflows
     before_action :samples, only: %i[create]
     before_action :workflow, only: %i[create]
 
     def pipeline_selection
+      @namespace_id = params[:namespace_id]
```

Comment:
Since you have `@namespace_id = params[:namespace_id]` multiple times, move the setting of `@namespace_id` into a method which is called using `before_action`  for the `create` and `pipeline_selection` actions